build/make.sh

```diff
@@ -1,8 +1,53 @@
 #!/bin/sh
-set +e
+set -euo pipefail
 cd $(dirname $(dirname $(realpath $0)))
-cd sox-src
 
-./configure
-make -s
-./src/sox -h
+# Detect platform
+platform=$(uname -s)
+
+if [ "$platform" = "Darwin" ]; then
+    echo "Building for macOS (Universal Binary)..."
+    
+    # Build for Intel (x86_64)
+    mkdir -p sox-build/x86_64
+    cd sox-src
+    CFLAGS="-arch x86_64 -Wno-incompatible-function-pointer-types" ./configure --prefix=$(pwd)/../sox-build/x86_64
+    make -s
+    make install
+    make clean
+
+    # Build for ARM (arm64)
+    mkdir -p sox-build/arm64
+    CFLAGS="-arch arm64 -Wno-incompatible-function-pointer-types" ./configure --prefix=$(pwd)/../sox-build/arm64
+    make -s
+    make install
```

Comment:
I think this script should not mutate the OS environment, so I'd prefer not to have the `make install` in here.